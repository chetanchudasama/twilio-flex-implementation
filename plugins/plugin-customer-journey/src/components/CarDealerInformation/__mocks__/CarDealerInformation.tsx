import React from "react";
import { Props } from "../CarDealerInformation";

const CarDealerInformation: React.FC<Props> = (props) => <div />;

export default CarDealerInformation;
