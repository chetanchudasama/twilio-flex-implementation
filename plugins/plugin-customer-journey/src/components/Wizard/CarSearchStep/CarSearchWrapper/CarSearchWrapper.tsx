import { BaseWizardStepProps, WizardStepWrapper } from "@common/components";
import { Grid, Tabs, Tab } from "@material-ui/core";
import * as Flex from "@twilio/flex-ui";

import React, { useState, useMemo, useEffect } from "react";
import { Manager } from "@twilio/flex-ui";

import SearchIcon from "@material-ui/icons/Search";
import FavoriteIcon from "@material-ui/icons/Favorite";
import { useApplicationService } from "../../../../services/application.service";
import { AppState } from "../../../../states";
import { CarSearchWrapperStyles } from "./CarSearchWrapper.Styles";
import { StateToProps, DispatchToProps } from "./CarSearchWrapper.Props";
import CarSearchResult from "../CarSearchResult/CarSearchResult.Container";
import { showErrorMessage } from "../../../../Notifications";
import CarSearchCriteria from "../CarSearchCriteria/CarSearchCriteria.Container";
import { useStaticService } from "../../../../services/static.service";
import { useVehicleService } from "../../../../services/vehicle.service";
import SavedCarsList from "../SavedCarsList/SavedCarsList.Container";
import { VehicleDetailModel } from "../../../../models/VehicleDetailModel";
import { VehicleSummaryDetailModel } from "../../../../models/VehicleSummaryDetailModel";

// eslint-disable-next-line @typescript-eslint/no-empty-interface
interface OwnProps {}

export type ComponentProps = StateToProps &
  DispatchToProps &
  OwnProps &
  BaseWizardStepProps;

const CarSearchWrapper: React.FC<ComponentProps> = ({
  applicationId,
  filters,
  hasSavedQuotes,
  savedVehicleCount,
  isVehicleStaticDataFetched,
  postCode,
  setVehicleStaticDataFetched,
  setVehicleSearchDropdownData,
  moveForward,
  moveBackward,
  setSavedVehicles,
  setSavedQuotes,
  setVehicles,
  setTotalVehicles,
  nextStep,
  previousStep,
  vrmFilters,
}) => {
  const state: AppState = Manager.getInstance().store.getState();
  const applicationService = useApplicationService(
    state.flex.session.ssoTokenPayload.token
  );
  const token: string = useMemo(() => {
    return state.flex.session.ssoTokenPayload.token ?? "";
  }, [state.flex.session.ssoTokenPayload]);

  const phoenixToken: string = useMemo(() => {
    console.log(
      "SETTING PHOENIX TOKEN MEMO WITH VALUE",
      state.crm?.crmState?.phoenixToken
    );
    return state.crm?.crmState?.phoenixToken ?? "";
  }, [state.crm?.crmState?.phoenixToken]);

  const staticService = useStaticService(token, phoenixToken);
  const vehicleService = useVehicleService(token, phoenixToken);

  const [currentTab, setTab] = useState<number>(0);

  const onNext = () => {
    if (moveForward) {
      moveForward();
    }
  };
  const onPrevious = () => {
    if (moveBackward) {
      moveBackward();
    }
  };

  useEffect(() => {
    if (applicationId <= 0) {
      return;
    }
    applicationService.getQuoteList(applicationId).then((response) => {
      setSavedQuotes(response);
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [applicationId]);

  useEffect(() => {
    if (applicationId <= 0) {
      return;
    }
    applicationService.getSavedVehicles(applicationId).then((response) => {
      setSavedVehicles(
        response.map((r) => Object.assign(new VehicleDetailModel(), r))
      );
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [applicationId]);

  useEffect(() => {
    // get vehicle search dropdown data
    if (isVehicleStaticDataFetched) {
      return;
    }

    staticService
      .getVehicleSearchDropdownData()
      .then((response) => {
        // TODO - decide where to return these from
        // Hard code distances for now
        response.distance = [1, 2, 3, 4, 5, 10, 15, 20, 30, 50, 100, 200, 300];

        setVehicleSearchDropdownData(response);
        setVehicleStaticDataFetched(true);
      })
      .catch((error) => {
        showErrorMessage("Something went wrong, please try again!", "", true);
      });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const searchVehicle = () => {
    // eslint-disable-next-line no-param-reassign
    filters.postcode = postCode;
    vehicleService
      .vehicleSearch(filters)
      .then((response) => {
        setVehicles(
          response.vehicles.map((v) =>
            Object.assign(new VehicleSummaryDetailModel(), v)
          )
        );
        setTotalVehicles(response.total);
      })
      .catch((error) => {
        showErrorMessage("Something went wrong, please try again!", "", true);
      });
  };

  useEffect(() => {
    if (filters.distance) {
      searchVehicle();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [filters]);

  useEffect(() => {
    Flex.Actions.registerAction(
      "MoveToCarSearchStep",
      async (payload: { tabIndex: number }) => {
        setTab(payload.tabIndex);
      }
    );
  }, []);

  // eslint-disable-next-line @typescript-eslint/ban-types
  const handleChange = (event: React.ChangeEvent<{}>, value: number) => {
    setTab(value);
  };

  const [height, setHeight] = useState("fit-content");

  useEffect(() => {
    let timeout: any = null;
    const calculateHeight = () => {
      // get twilio header height
      const twilioHeading =
        document.getElementsByClassName("Twilio-MainHeader");
      let twilioHeadingHeight = 0;
      if (twilioHeading && twilioHeading.length) {
        twilioHeadingHeight = (twilioHeading[0] as HTMLElement).offsetHeight;
      }
      const customerBanner = document.getElementsByClassName("customer-banner");
      let customerBannerHeight = 0;
      if (customerBanner && customerBanner.length) {
        customerBannerHeight = (customerBanner[0] as HTMLElement).offsetHeight;
      }

      const actionContainer =
        document.getElementsByClassName("actionContainer");
      let actionContainerHeight = 0;
      if (actionContainer && actionContainer.length) {
        actionContainerHeight = (actionContainer[0] as HTMLElement)
          .offsetHeight;
      }

      const stepper = document.getElementsByClassName("stepper");
      let stepperHeight = 0;
      if (stepper && stepper.length) {
        stepperHeight = (stepper[0] as HTMLElement).offsetHeight;
      }

      const tabContainer = document.getElementsByClassName("tab-container");
      let tabContainerHeight = 0;
      if (tabContainer && tabContainer.length) {
        tabContainerHeight = (tabContainer[0] as HTMLElement).offsetHeight;
      }

      timeout = setTimeout(() => {
        let h = "fit-content";
        if (
          twilioHeadingHeight > 0 ||
          customerBannerHeight > 0 ||
          actionContainerHeight > 0 ||
          stepperHeight > 0 ||
          tabContainerHeight > 0
        ) {
          h = `${
            window.innerHeight -
            twilioHeadingHeight -
            customerBannerHeight -
            actionContainerHeight -
            stepperHeight -
            tabContainerHeight -
            15
          }px`;
        }
        if (h !== height) {
          setHeight(h);
        }
      }, 0);
    };
    // update height on window resize
    document.addEventListener("pointermove", calculateHeight);
    calculateHeight();
    return () => {
      clearTimeout(timeout);
      document.removeEventListener("pointermove", calculateHeight);
    };
  }, [height, currentTab]);

  useEffect(() => {
    if (vrmFilters.vrm) {
      vehicleService
        .vehicleSearchByVRM(vrmFilters)
        .then((response) => {
          if (response.vehicles && response.vehicles.length > 0) {
            setVehicles(
              response.vehicles.map((v) =>
                Object.assign(new VehicleSummaryDetailModel(), v)
              )
            );
          }
          setTotalVehicles(response.total);
        })
        .catch(() => {
          showErrorMessage("Something went wrong, please try again!", "", true);
        });
    }
  }, [setTotalVehicles, setVehicles, vehicleService, vrmFilters]);

  return (
    <CarSearchWrapperStyles>
      <WizardStepWrapper
        onNext={onNext}
        onPrevious={onPrevious}
        disableNext={!hasSavedQuotes}
        classes={{
          content: "wizard-content",
        }}
        nextStep={nextStep}
        previousStep={previousStep}
      >
        <Grid item xs={12}>
          <Grid container direction="row" justify="center" alignItems="center">
            <Grid item xs={12} className="tab-container">
              <Tabs
                value={currentTab}
                onChange={handleChange}
                indicatorColor="secondary"
                textColor="secondary"
                centered
              >
                <Tab
                  value={0}
                  label={
                    <>
                      <SearchIcon className="tab-icon" />
                      <span className="tab-title">car search</span>
                    </>
                  }
                />
                <Tab
                  value={1}
                  label={
                    <>
                      <FavoriteIcon className="tab-icon" />
                      <span className="tab-title">{`Saved Cars (${savedVehicleCount})`}</span>
                    </>
                  }
                />
              </Tabs>
            </Grid>
            <Grid item xs={12} className="tab-content" style={{ height }}>
              {currentTab === 0 && (
                <Grid
                  container
                  direction="row"
                  justify="center"
                  alignItems="center"
                  spacing={16}
                >
                  <Grid item xs={12}>
                    <CarSearchCriteria />
                  </Grid>
                  <Grid item xs={12}>
                    <CarSearchResult />
                  </Grid>
                </Grid>
              )}
              {currentTab === 1 && (
                <Grid item xs={12}>
                  <SavedCarsList />
                </Grid>
              )}
            </Grid>
          </Grid>
        </Grid>
      </WizardStepWrapper>
    </CarSearchWrapperStyles>
  );
};

export default CarSearchWrapper;
