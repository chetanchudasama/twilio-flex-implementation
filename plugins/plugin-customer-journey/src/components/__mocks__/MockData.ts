import { WorkerAttributes } from "@twilio/flex-ui";
import { namespace } from "../../states";

const workerAttributes: WorkerAttributes = {
  contact_uri: "client:test",
  email: "test@test.com",
  full_name: "test user",
  image_url: "img_url",
  roles: ["admin"],
  agentId: 1,
};

const conferenceSource = {
  source: {
    participants: [
      {
        callSid: "callSid",
        participantType: "customer",
        workerSid: "",
      },
      {
        callSid: "callSid",
        participantType: "worker",
        workerSid: "workerSid",
      },
    ],
  },
};

const conferenceState = new Map();
conferenceState.set("taskSid", conferenceSource);

export const flexManagerMock = jest.fn().mockReturnValue({
  workerClient: {
    source: {},
    sid: "workerSid",
    name: "test user",
    fullName: "test user",
    activityName: "Offline",
    attributes: workerAttributes,
    dateUpdated: new Date(),
    isAvailable: false,
    activityDuration: "",
    dateStatusChanged: new Date(1).toISOString(),
    activity: {
      available: false,
      name: "Offline",
      sid: "OfflineSid",
    },
  },
  store: {
    getState: jest.fn().mockReturnValue({
      flex: {
        worker: {
          attributes: workerAttributes,
        },
        session: {
          ssoTokenPayload: {
            token: "ssoToken",
          },
        },
        conferences: {
          states: conferenceState,
        },
      },
      [namespace]: {
        phoneNumberList: {
          phoneNumberList: [
            {
              friendlyName: "Sales",
              phoneNumber: "+1234567891",
              queueSid: "QUEUE_SID",
            },
          ],
        },
      },
    }),
  },
  strings: {
    DIALPADExternalTransferSearchInputLabel: "Search By Name or PhoneNumber",
    DIALPADExternalTransferHoverOver: "Add",
    DIALPADExternalTransferPopupHeader: "External Transfer",
  },
  serviceConfiguration: {
    outbound_call_flows: {
      default: {
        caller_id: "CALLER_ID",
      },
    },
  },
});
