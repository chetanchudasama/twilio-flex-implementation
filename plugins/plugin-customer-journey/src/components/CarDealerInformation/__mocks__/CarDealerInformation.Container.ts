import CarDealerInformation from "./CarDealerInformation";

export default CarDealerInformation;
