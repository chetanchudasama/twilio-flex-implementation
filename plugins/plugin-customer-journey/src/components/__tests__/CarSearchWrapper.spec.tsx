import { mount } from "enzyme";
import flushPromises from "flush-promises";
import React from "react";
import { act } from "react-dom/test-utils";
import { create } from "react-test-renderer";

import { ItemModel } from "@common/components";
import { Manager } from "@twilio/flex-ui";

import { VehicleVRMSearchRequestModel } from "../../models/VehicleVRMSearchRequestModel";
import { QuoteDetailModel } from "../../models/QuoteDetailModel";
import { VehicleDetailModel } from "../../models/VehicleDetailModel";
import { VehicleSearchRequestModel } from "../../models/VehicleSearchRequestModel";
import { VehicleSummaryDetailModel } from "../../models/VehicleSummaryDetailModel";
import { flexManagerMock } from "../__mocks__/MockData";
import CarSearchWrapper, {
  ComponentProps,
} from "../Wizard/CarSearchStep/CarSearchWrapper/CarSearchWrapper";

jest.mock("../../services/static.service");
jest.mock("../../services/vehicle.service");
jest.mock("../../services/application.service");
jest.mock(
  "../Wizard/CarSearchStep/CarSearchCriteria/CarSearchCriteria.Container"
);
jest.mock("../Wizard/CarSearchStep/CarSearchResult/CarSearchResult.Container");
jest.mock("../Wizard/CarSearchStep/CarSearchResult/CarSearchResult");

const mockGetDropdownData = jest.fn();
const mockVehicleSearch = jest.fn();
const mockGetQuoteList = jest.fn();
const mockShowErrorMessage = jest.fn();
const mockGetSavedVehicles = jest.fn();
const mockUpdateSavedVehicles = jest.fn();
const mockVehicleSearchByVRM = jest.fn();

jest.mock("../../services/static.service", () => ({
  useStaticService: jest.fn(() => ({
    getVehicleSearchDropdownData: mockGetDropdownData,
  })),
}));

jest.mock("../../services/vehicle.service", () => ({
  useVehicleService: jest.fn(() => ({
    vehicleSearch: mockVehicleSearch,
    vehicleSearchByVRM: mockVehicleSearchByVRM,
  })),
}));

jest.mock("../../services/application.service", () => ({
  useApplicationService: jest.fn(() => ({
    getQuoteList: mockGetQuoteList,
    getSavedVehicles: mockGetSavedVehicles,
    updateSavedVehicles: mockUpdateSavedVehicles,
  })),
}));

jest.mock("../../Notifications", () => ({
  showErrorMessage: jest.fn(() => mockShowErrorMessage),
}));

jest.mock("@common/components", () => {
  const components = jest.requireActual("@common/components");
  return {
    ...components,
    WizardStepWrapper: jest.fn(({ children, onNext, onPrevious }) => {
      return (
        <div>
          <button type="button" className="onNext" onClick={onNext}>
            onNext
          </button>
          <button type="button" className="onPrevious" onClick={onPrevious}>
            onPrevious
          </button>
          {children}
        </div>
      );
    }),
  };
});

Manager.getInstance = flexManagerMock;

const defaultProps: ComponentProps = {
  postCode: "",
  applicationId: 1,
  filters: new VehicleSearchRequestModel(),
  isVehicleStaticDataFetched: true,
  hasSavedQuotes: false,
  savedVehicleCount: 0,
  setVehicleSearchDropdownData: jest.fn(() => {}),
  setVehicleStaticDataFetched: jest.fn(() => {}),
  setVehicles: jest.fn(() => {}),
  setSavedVehicles: jest.fn(() => {}),
  setSavedQuotes: jest.fn(() => {}),
  setTotalVehicles: jest.fn(() => {}),
  vrmFilters: new VehicleVRMSearchRequestModel(),
};

const items: ItemModel[] = [
  { id: 1, name: "test1" },
  { id: 2, name: "test2" },
  { id: 3, name: "test3" },
];

const dropdownData = {
  make: [
    {
      id: 1,
      name: "test1",
      models: [
        { id: 1, name: "test1", count: 1 },
        { id: 2, name: "test2", count: 2 },
      ],
    },
    {
      id: 2,
      name: "test2",
      models: [
        { id: 1, name: "test1", count: 1 },
        { id: 2, name: "test2", count: 2 },
      ],
    },
    {
      id: 3,
      name: "test3",
      models: [
        { id: 1, name: "test1", count: 1 },
        { id: 2, name: "test2", count: 2 },
      ],
    },
  ],
  bodyType: items,
  colour: items,
  fuelType: items,
  transmission: items,
  distance: [2, 5, 7],
};

const searchResultData = {
  page: 0,
  pageSize: 0,
  total: 0,
  vehicles: [new VehicleSummaryDetailModel()],
};

let props: ComponentProps = defaultProps;

Manager.getInstance = flexManagerMock;

describe("CarSearchWrapper", () => {
  beforeEach(() => {
    props = defaultProps;
    mockGetDropdownData.mockResolvedValueOnce(() =>
      Promise.resolve(dropdownData)
    );
    mockGetQuoteList.mockResolvedValueOnce(() =>
      Promise.resolve([new QuoteDetailModel()])
    );
    mockVehicleSearch.mockResolvedValueOnce(() =>
      Promise.resolve(searchResultData)
    );
    mockGetSavedVehicles.mockImplementation(() =>
      Promise.resolve([new VehicleDetailModel()])
    );
    mockVehicleSearchByVRM.mockResolvedValueOnce(() =>
      Promise.resolve(searchResultData)
    );
  });

  afterEach(() => {
    mockGetDropdownData.mockClear();
    mockGetQuoteList.mockClear();
    mockVehicleSearch.mockClear();
    mockGetSavedVehicles.mockClear();
    mockVehicleSearchByVRM.mockClear();

    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  it("renders correctly, without any props", async () => {
    const wrapper = create(<CarSearchWrapper {...props} />).toJSON();
    await flushPromises();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly, with props", async () => {
    props.moveForward = jest.fn();
    props.moveBackward = jest.fn();
    const wrapper = create(<CarSearchWrapper {...props} />).toJSON();
    await flushPromises();
    expect(wrapper).toMatchSnapshot();
  });

  it("when next button clicked and move forward event is provided, move forward event should be emitted", async () => {
    const moveForward = jest.fn();
    props.moveForward = moveForward;
    const wrapper = mount(<CarSearchWrapper {...props} />);
    await flushPromises();
    wrapper.find(".onNext").last().simulate("click");

    expect(moveForward).toHaveBeenCalledTimes(1);
  });

  it("when previous button clicked and move backward event is provided, move backward event should be emitted", async () => {
    const moveBackward = jest.fn();
    props.moveBackward = moveBackward;
    const wrapper = mount(<CarSearchWrapper {...props} />);
    await flushPromises();
    wrapper.find(".onPrevious").last().simulate("click");

    expect(moveBackward).toHaveBeenCalledTimes(1);
  });

  it("should fetch dropdownData if not fetched", async () => {
    const moveBackward = jest.fn();
    props.moveBackward = moveBackward;
    props.isVehicleStaticDataFetched = false;

    mockGetDropdownData.mockImplementationOnce(() =>
      Promise.resolve(dropdownData)
    );
    mockVehicleSearch.mockImplementationOnce(() =>
      Promise.resolve(searchResultData)
    );

    await act(async () => {
      const wrapper = mount(<CarSearchWrapper {...props} />);
      await flushPromises();
      wrapper.update();

      expect(mockGetDropdownData).toHaveBeenCalledTimes(1);
    });
  });

  it("should not fetch dropdownData if data already stored in store", async () => {
    const moveBackward = jest.fn();
    props.moveBackward = moveBackward;
    props.isVehicleStaticDataFetched = true;

    mockGetDropdownData.mockImplementationOnce(() =>
      Promise.resolve(dropdownData)
    );

    await act(async () => {
      const wrapper = mount(<CarSearchWrapper {...props} />);
      await flushPromises();
      wrapper.update();

      expect(mockGetDropdownData).not.toHaveBeenCalled();
    });
  });

  it("when search button is clicked, vehicleSearch API should be called", async () => {
    const moveBackward = jest.fn();
    props.moveBackward = moveBackward;
    props.postCode = "test";
    props.filters = Object.assign(new VehicleSearchRequestModel(), {
      distance: 3,
    });

    mockVehicleSearch.mockImplementationOnce(() =>
      Promise.resolve(searchResultData)
    );

    await act(async () => {
      const wrapper = mount(<CarSearchWrapper {...props} />);
      await flushPromises();
      wrapper.update();

      expect(mockVehicleSearch).toHaveBeenCalled();
    });
  });

  it("should fetch quote list when application id is present", async () => {
    props.applicationId = 1;

    mockGetQuoteList.mockImplementationOnce(() =>
      Promise.resolve(new QuoteDetailModel())
    );
    mockVehicleSearch.mockImplementationOnce(() =>
      Promise.resolve(searchResultData)
    );

    await act(async () => {
      const wrapper = mount(<CarSearchWrapper {...props} />);
      wrapper.update();
      await flushPromises();
      wrapper.update();
    });

    expect(mockGetQuoteList).toHaveBeenCalledTimes(1);
  });

  it("should not fetch quote list if application id is not present", async () => {
    props.applicationId = 0;

    mockGetQuoteList.mockImplementationOnce(() =>
      Promise.resolve([new QuoteDetailModel()])
    );

    await act(async () => {
      const wrapper = mount(<CarSearchWrapper {...props} />);
      wrapper.update();
      await flushPromises();
      wrapper.update();
    });

    expect(mockGetQuoteList).not.toHaveBeenCalled();
  });

  it("should fetch saved cars, when application id is present", async () => {
    props.applicationId = 1;

    mockVehicleSearch.mockImplementationOnce(() =>
      Promise.resolve(searchResultData)
    );

    await act(async () => {
      const wrapper = mount(<CarSearchWrapper {...props} />);
      wrapper.update();
      await flushPromises();
      wrapper.update();
    });

    expect(mockGetSavedVehicles).toHaveBeenCalledTimes(1);
  });

  it("when search by vrm button is clicked, vehicleSearchByVRM API should be called", async () => {
    const moveBackward = jest.fn();
    props.moveBackward = moveBackward;
    props.filters.distance = null;
    props.vrmFilters = Object.assign(new VehicleVRMSearchRequestModel(), {
      vrm: "test",
    });

    mockVehicleSearchByVRM.mockImplementationOnce(() =>
      Promise.resolve(searchResultData)
    );

    await act(async () => {
      const wrapper = mount(<CarSearchWrapper {...props} />);
      await flushPromises();
      wrapper.update();

      expect(mockVehicleSearchByVRM).toHaveBeenCalled();
    });
  });
});
