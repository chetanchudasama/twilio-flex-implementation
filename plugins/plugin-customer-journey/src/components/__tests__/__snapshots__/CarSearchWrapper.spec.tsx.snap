// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarSearchWrapper renders correctly, with props 1`] = `
<div
  className="css-6e08cn"
>
  <div>
    <button
      className="onNext"
      onClick={[Function]}
      type="button"
    >
      onNext
    </button>
    <button
      className="onPrevious"
      onClick={[Function]}
      type="button"
    >
      onPrevious
    </button>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
    >
      <div
        className="MuiGrid-container-1 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
      >
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 tab-container"
        >
          <div
            className="MuiTabs-root-98"
          >
            <div
              className="MuiTabs-flexContainer-99"
            >
              <div
                className="MuiTabs-scroller-101 MuiTabs-fixed-102"
                onScroll={[Function]}
                role="tablist"
                style={
                  Object {
                    "marginBottom": 0,
                  }
                }
              >
                <div
                  className="MuiTabs-flexContainer-99 MuiTabs-centered-100"
                >
                  <button
                    aria-selected={true}
                    className="MuiButtonBase-root-119 MuiTab-root-107 MuiTab-textColorSecondary-111 MuiTab-selected-112"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="tab"
                    tabIndex="0"
                    type="button"
                  >
                    <span
                      className="MuiTab-wrapper-115"
                    >
                      <span
                        className="MuiTab-labelContainer-116"
                      >
                        <span
                          className="MuiTab-label-117"
                        >
                          <svg
                            aria-hidden="true"
                            className="MuiSvgIcon-root-122 tab-icon"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                          <span
                            className="tab-title"
                          >
                            car search
                          </span>
                        </span>
                      </span>
                    </span>
                    <span
                      className="MuiTouchRipple-root-134"
                    />
                  </button>
                  <button
                    aria-selected={false}
                    className="MuiButtonBase-root-119 MuiTab-root-107 MuiTab-textColorSecondary-111"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="tab"
                    tabIndex="0"
                    type="button"
                  >
                    <span
                      className="MuiTab-wrapper-115"
                    >
                      <span
                        className="MuiTab-labelContainer-116"
                      >
                        <span
                          className="MuiTab-label-117"
                        >
                          <svg
                            aria-hidden="true"
                            className="MuiSvgIcon-root-122 tab-icon"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                            <path
                              d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                            />
                          </svg>
                          <span
                            className="tab-title"
                          >
                            Saved Cars (0)
                          </span>
                        </span>
                      </span>
                    </span>
                    <span
                      className="MuiTouchRipple-root-134"
                    />
                  </button>
                </div>
                <span
                  className="MuiPrivateTabIndicator-root-131 MuiPrivateTabIndicator-colorSecondary-133 MuiTabs-indicator-106"
                  style={
                    Object {
                      "left": 0,
                      "width": 0,
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 tab-content"
          style={
            Object {
              "height": "fit-content",
            }
          }
        >
          <div
            className="MuiGrid-container-1 MuiGrid-spacing-xs-16-24 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
          >
            <div
              className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
            >
              <div>
                <button
                  className="search-btn"
                  onClick={[Function]}
                  type="button"
                >
                  Search
                </button>
              </div>
            </div>
            <div
              className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
            >
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CarSearchWrapper renders correctly, without any props 1`] = `
<div
  className="css-6e08cn"
>
  <div>
    <button
      className="onNext"
      onClick={[Function]}
      type="button"
    >
      onNext
    </button>
    <button
      className="onPrevious"
      onClick={[Function]}
      type="button"
    >
      onPrevious
    </button>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
    >
      <div
        className="MuiGrid-container-1 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
      >
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 tab-container"
        >
          <div
            className="MuiTabs-root-98"
          >
            <div
              className="MuiTabs-flexContainer-99"
            >
              <div
                className="MuiTabs-scroller-101 MuiTabs-fixed-102"
                onScroll={[Function]}
                role="tablist"
                style={
                  Object {
                    "marginBottom": 0,
                  }
                }
              >
                <div
                  className="MuiTabs-flexContainer-99 MuiTabs-centered-100"
                >
                  <button
                    aria-selected={true}
                    className="MuiButtonBase-root-119 MuiTab-root-107 MuiTab-textColorSecondary-111 MuiTab-selected-112"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="tab"
                    tabIndex="0"
                    type="button"
                  >
                    <span
                      className="MuiTab-wrapper-115"
                    >
                      <span
                        className="MuiTab-labelContainer-116"
                      >
                        <span
                          className="MuiTab-label-117"
                        >
                          <svg
                            aria-hidden="true"
                            className="MuiSvgIcon-root-122 tab-icon"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                          <span
                            className="tab-title"
                          >
                            car search
                          </span>
                        </span>
                      </span>
                    </span>
                    <span
                      className="MuiTouchRipple-root-134"
                    />
                  </button>
                  <button
                    aria-selected={false}
                    className="MuiButtonBase-root-119 MuiTab-root-107 MuiTab-textColorSecondary-111"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="tab"
                    tabIndex="0"
                    type="button"
                  >
                    <span
                      className="MuiTab-wrapper-115"
                    >
                      <span
                        className="MuiTab-labelContainer-116"
                      >
                        <span
                          className="MuiTab-label-117"
                        >
                          <svg
                            aria-hidden="true"
                            className="MuiSvgIcon-root-122 tab-icon"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                            <path
                              d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                            />
                          </svg>
                          <span
                            className="tab-title"
                          >
                            Saved Cars (0)
                          </span>
                        </span>
                      </span>
                    </span>
                    <span
                      className="MuiTouchRipple-root-134"
                    />
                  </button>
                </div>
                <span
                  className="MuiPrivateTabIndicator-root-131 MuiPrivateTabIndicator-colorSecondary-133 MuiTabs-indicator-106"
                  style={
                    Object {
                      "left": 0,
                      "width": 0,
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 tab-content"
          style={
            Object {
              "height": "fit-content",
            }
          }
        >
          <div
            className="MuiGrid-container-1 MuiGrid-spacing-xs-16-24 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
          >
            <div
              className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
            >
              <div>
                <button
                  className="search-btn"
                  onClick={[Function]}
                  type="button"
                >
                  Search
                </button>
              </div>
            </div>
            <div
              className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
            >
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
